### DISCUSSION TITLE
Can I add an Onion feed in FreshRSS ?

### DISCUSSION THREAD

**Discussion by niladrishekhar** [FreshRSS/FreshRSS#3536] [Q&A]
I have a few onion tor based sites which offer RSS feeds. How can I add them in my new feed. 
I tried adding one changing the proxy type to SOCKS5 and the port as localhost:9050 but I was not able to add the feed - there was an error. 
cURL error 6: Could not resolve host: rssfeedr622fbf3tgg4vxp43hfhax76rqzifhj2j43j6xqjmier7z7id.onion [http://rssfeedr622fbf3tgg4vxp43hfhax76rqzifhj2j43j6xqjmier7z7id.onion/]

How can I add one ?

**Comment by Alkarex** [comment]
Hello,
Can you cURL it from your server? If yes, that should be possible.

**Reply by niladrishekhar** [reply]
This is what I tried - 
1. Started the tor service using - sudo systemctl start tor
2. tested if tor is working - curl --socks5 localhost:9050 --socks5-hostname localhost:9050 -s https://check.torproject.org/ | cat | grep -m 1 Congratulations | xargs         result shown - Congratulations. This browser is configured to use Tor.
3. curl --socks5-hostname localhost:9050 http://www.google.com/    result shown - gives me the content 
4. trying for an onion site curl --socks5 localhost:9050 https://rss.nytimes3xbfgragh.onion/services/xml/rss/nyt/HomePage.xml
gives me - curl: (97) Could not resolve host: rss.nytimes3xbfgragh.onion

Alternatively tried the following as well 
1. curl --socks5-hostname localhost:9050 https://rss.nytimes3xbfgragh.onion/services/xml/rss/nyt/HomePage.xml gives curl: (97) Can't complete SOCKS5 connection to rss.nytimes3xbfgragh.onion. (6)
2. 
2. curl -s --socks5-hostname localhost:9050 https://rss.nytimes3xbfgragh.onion/services/xml/rss/nyt/HomePage.xml
gives no error message and no content either. (as -s silences the curl output)

Am I working in the right direction ?

**Reply by Alkarex** [reply]
During the debugging of your cURL commands, use a `-v` (verbose) and not a `-s` (silent). Yes, that is the right direction. If you can get a cURL to work, we should be able to move it to FreshRSS.

**Reply by niladrishekhar** [reply]
Hi Alkarex, Thanks for the guidance I was able to make it work - 
curl -o log.txt -v --socks5-hostname localhost:9050 https://rss.nytimes3xbfgragh.onion/services/xml/rss/nyt/HomePage.xml

Attaching the files for your reference. However when I tried to add the feed I selected SOCKS5 from drop down and in the localhost gave 127.0.0.1:9050 the feed wasn't added.

Log error - 
cURL error 97: Could not resolve host: rss.nytimes3xbfgragh.onion 
The file log.txt is basically the output of the cURL -o. Many thanks in advance. 

[https://rss.nytimes3xbfgragh.onion/services/xml/rss/nyt/HomePage.xml]
[cURL Terminal Output.txt](https://github.com/FreshRSS/FreshRSS/files/6164868/cURL.Terminal.Output.txt)
[log.txt](https://github.com/FreshRSS/FreshRSS/files/6164870/log.txt)

**Comment by Saklad5** [comment]
cURL allows you to specify whether resolution happens through a proxy as well: that CLI example says it should, with the "hostname" part of `--socks5-hostname`, but that doesn't necessarily mean FreshRSS knows that.

**Comment by bocciaman** [comment]
Is there a way to add news sites you are subscribed to on FreshRSS with full-feed text from [Fivefilters](https://www.fivefilters.org)? News sites like wsj.com for example?

**Reply by Frenzie** [reply]
I don't use it, but they list RSS (and JSON) feed creation as one of their features, so I don't see why not?

**Comment by ghost** [comment]


**Reply by Alkarex** [reply]
Could you try to add that feed with the `SOCKS5H` option of FreshRSS?

**Reply by Alkarex** [reply]
Thanks for the feedback 👍🏻  I believe instead of `docker exec`, you could try to install the same thing in another independent container. The only thing to change would be to use the name of that container instead of `localhost`, and making sure that this container is on the same Docker network than FreshRSS.
That way the setup would not be broken at each FreshRSS container update.
It would be nice to consolidate that documentation here or elsewhere.

**Reply by Alkarex** [reply]
Great progress 👍🏻 There should be no need to search the IP address of the container. The idea is to use the container name instead, like `socks5h://tor-proxy:9050`

**Reply by Alkarex** [reply]
Excellent. Now a new improvement would be to put that in a Docker Compose example like https://github.com/FreshRSS/FreshRSS/blob/edge/Docker/freshrss/docker-compose-db.yml (that is an example for a dedicated database).
So it may be run like e.g.:

```sh
docker compose -f docker-compose.yml -f docker-compose-tor.yml up
```

See https://github.com/FreshRSS/FreshRSS/blob/edge/Docker/README.md#docker-compose

Simplifying some things, in particular no need to deal with networks manually.

**Reply by Alkarex** [reply]
You do not need `ports` at all for the `tor-proxy` container if it is only supposed to be accessed from FreshRSS or other containers of the same Docker network
